lib/binaries/iedriver_xml.ts

```diff
@@ -0,0 +1,82 @@
+import * as semver from 'semver';
+
+import {Config} from '../config';
+
+import {BinaryUrl} from './binary';
+import {XmlConfigSource} from './config_source';
+
+export class IEDriverXml extends XmlConfigSource {
+  constructor() {
+    super('iedriver');
+    this.xmlUrl = Config.cdnUrls()['ie'];
+  }
+
+  getUrl(version: string): Promise<BinaryUrl> {
+    return this.getXml().then(() => {
+      if (version === 'latest') {
+        return this.getLatestIEDriverVersion();
+      } else {
+        return this.getSpecificIEDriverVersion(version);
+      }
+    });
+  }
+
+  getVersionList(): Promise<string[]> {
+    return this.getXml().then(() => {
+      return this.getIEDriverList();
+    });
+  }
+
+
+  private getIEDriverList(): string[] {
+    let versionPaths: string[] = [];
+
+    for (let content of this.xml.ListBucketResult.Contents) {
+      let contentKey: string = content.Key[0];
+
+      // Filter For IEDriverServer win 32. Removing option to download x64
+      if (contentKey.includes('IEDriverServer_Win32_')) {
+        versionPaths.push(contentKey);
+      }
+    }
+    return versionPaths;
+  }
+
+  private getLatestIEDriverVersion(): BinaryUrl {
```

Comment:
I'd rather leave this as is. Getting the latest is different in chrome, selenium, ie, and gecko. To generalize this would make things more complicated for future work.